### DISCUSSION TITLE
Is there a way to read variables from the local .env file ?

### DISCUSSION THREAD

**Discussion by HassanZahirnia** [deployphp/deployer#3479] [Help needed]
I'd rather not have my host configurations such as host name, port, username, paths, etc... in the `deploy.php` and I prefer having them in a config file that is gitignored.
I wonder if there is a way to read variables from a `.env` file ?

**Comment by antonmedv** [comment]
Yes, via Dotenv.

**Reply by HassanZahirnia** [reply]
Can you please provide an example? I could not get it working. I keep getting `Class "Dotenv\Dotenv" not found` error

**Comment by beyto1974** [comment]
Hi,

This happens when you install deployer.org using the binary or globally. 

In the deployer folder run:
```bash
composer require deployer/deployer
composer require vlucas/phpdotenv
```

Example code:
```php
require __DIR__ . '/vendor/autoload.php';

$dotenv = \Dotenv\Dotenv::createImmutable(__DIR__);
$dotenv->safeLoad();

set('slack_webhook', getenv('SLACK_WEBHOOK'));
set('slack_channel', '#dev');
set('slack_title', 'Deployment');
set('slack_success_text', 'Deploy successful');
set('slack_failure_text', 'Deploy failed');
```

This works, but I couldn't get it working with optional variables:

```php
if(getenv('SLACK_WEBHOOK')){
   set('slack_webhook', getenv('SLACK_WEBHOOK'));
   // ...
}
```

The notification is never sent.
